xnmt/hyper_params.py

```diff
@@ -72,3 +72,8 @@ def __init__(self, sequence=None, initial=0.0, update=0):
   def update_value(self):
     return self.sequence[min(len(self.sequence)-1, self.update)]
 
+  def __hash__(self):
```

Comment:
In #523 I had also been thinking about this, but learned that hash function must always return the same value. Hyper params are mutable, so I think we shouldn't implement a hash function here. Is there a particular reason you need this?

See also: https://hynek.me/articles/hashes-and-equality/

Reply:
Actually, this was a quick hack to make my experiment running. Previously I have problem when serializing the hyperparams. Some code in the serialization tried to add the HyperParam to set() and leads to unhashable error. Currently I am still facing the serialization problem but other one. I will make a unittest and let you know if I can't fix that.

Reply:
Okay yeah, I had encountered the same problem but it should already be fixed by #523.